### PR TITLE
Remove DOMPurify and refactor Sanitizer API

### PR DESCRIPTION
**Pull request by choumx**
- Removes DOMPurify from worker-dom. Instead, require callers to provide an object that implements the `Sanitizer` interface in the `WorkerDOMConfiguration` object.
- Made some Sanitizer API changes to (1) include attribute removals and (2) delegate actual mutation to caller to support value rewriting.

### CODE REVIEW

File: src/main-thread/index.safe.ts [hunk 1]
```diff
@@ -14,36 +14,17 @@
...
-
-/** Users can import this and configure the sanitizer with custom DOMPurify hooks, etc. */
-export const sanitizer = new DOMPurifySanitizer();
-
 /**
- * @param baseElement
- * @param domURL
+ * @fileoverview This entry point API in active development and unstable.
  */
-export function upgradeElement(baseElement: Element, domURL: string, longTask?: LongTaskFunction): Promise<Worker | null> {
```
Comment (dvoytenko) [inline src/main-thread/index.safe.ts]: @kristoferbaxter was explaining to me that there's a key difference with this method where `domURL` actually plays a part. He wanted to remove `domURL` from the config and instead put it here as an argument.

File: src/worker-thread/index.safe.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ const WHITELISTED_GLOBALS = [
   'encodeURIComponent',
   'escape',
   'eval',
-  'fetch', // TODO(choumx): Wrap this for origin whitelisting.
```
Comment (dvoytenko) [inline src/worker-thread/index.safe.ts]: Are we skipping the "origin whitelisting" work?

File: src/main-thread/DOMPurifySanitizer.ts [hunk 3]
```diff
@@ -1,109 +0,0 @@
...
- * DOMPurify hooks before/after sanitization.
- * @see https://github.com/cure53/DOMPurify#hooks
- */
-interface DOMPurifySanitizerCallbacks {
-  afterSanitize?: (purify: DOMPurify) => void;
-  beforeSanitize?: (purify: DOMPurify) => void;
-  nodeWasRemoved?: (node: Node) => void;
-}
-
-export class DOMPurifySanitizer implements Sanitizer {
```
Comment (dvoytenko) [inline src/main-thread/DOMPurifySanitizer.ts]: This is an interesting situation. Can we actually setup an interface implementation over the boundary between AMP's Closure codebase and WorkerDOM's TypeScript? Can we even export `Sanitizer` interface over to Closure?

Comment (choumx) [reply src/main-thread/DOMPurifySanitizer.ts]: I don't think so, unless we export a types declaration from worker-dom and convert amp-script to TS. 

I actually added support for extensions in TS and transpiled to Closure-annotated JS via tsickle awhile ago. I'll file an issue for this.

Comment (dvoytenko) [reply src/main-thread/DOMPurifySanitizer.ts]: Ok. Should I rereview? Or not yet?

File: src/worker-thread/index.safe.ts (hunk 2 above)
Comment (choumx) [reply src/worker-thread/index.safe.ts]: Update: Tested this and looks like enforcement via `<meta>` CSP is possible. Will follow up with this separately.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#Multiple_content_security_policies

**Pull request merged**